makefile

```diff
@@ -0,0 +1,40 @@
+CC      := clang++
+FLAGS 	:= -Wall -Werror -Wshadow -Wextra -std=c++11 -ggdb3 -DDEBUG 
+LINK  	:= -lSDL2 -pthread
+OBJ_DIR := ./objects
+EXE   	:= game
+SRC     := $(wildcard *.cpp) \
+
+OBJECTS := $(SRC:%.cpp=$(OBJ_DIR)/%.o)
+
+# Tested and working on mac, untested on windows and linux
+ifeq ($(OS),Windows_NT)
+	LINK += -lGL
+else
+	UNAME_S := $(shell uname -s)
+	ifeq ($(UNAME_S),Linux)
+		LINK += -lGL
+
+	endif
+	ifeq ($(UNAME_S),Darwin)
+		LINK += -framework OpenGL
+
+	endif
+endif
+
+all: $(EXE)
+
+$(EXE): $(OBJECTS)
+	$(CC) -o $(EXE) $^ $(LINK)
+
+$(OBJ_DIR)/%.o: %.cpp
+	@mkdir -p $(@D)
+	$(CC) $(FLAGS) -c $< -o $@
+
+.PHONY: all clean release
+
+release: FLAGS += -O3
```

Comment:
While we certainly only want optimizations for the release build, FLAGS also already contain -DDEBUG and -ggdb3 by default, so we may want to create a separate debug target or something that only adds those flags ? (I haven't worked with makefiles in forever haha)

Reply:
Didn't think of that! I will make a debug target, but list it first so it's the default